src/pandas_profiling/model/typeset.py

```diff
@@ -0,0 +1,231 @@
+import imghdr
+import os
+import warnings
+from urllib.parse import urlparse
+
+import pandas as pd
+import visions
+from pandas.api import types as pdt
+from visions import VisionsBaseType, VisionsTypeset
+from visions.relations import IdentityRelation, InferenceRelation
+from visions.utils import nullable_series_contains
+
+from pandas_profiling.config import config
+from pandas_profiling.model.typeset_relations import (
+    category_is_numeric,
+    category_to_numeric,
+    numeric_is_category,
+    series_is_string,
+    string_is_bool,
+    string_to_bool,
+    to_bool,
+    to_category,
+)
+
+
+class PandasProfilingBaseType(VisionsBaseType):
+    pass
+
+
+class Unsupported(PandasProfilingBaseType, visions.Generic):
+    pass
+
+
+class Numeric(PandasProfilingBaseType):
+    @classmethod
+    def get_relations(cls):
+        return [
+            IdentityRelation(cls, Unsupported),
+            InferenceRelation(
+                cls,
+                Categorical,
+                relationship=category_is_numeric,
+                transformer=category_to_numeric,
+            ),
+        ]
+
+    @classmethod
+    @nullable_series_contains
+    def contains_op(cls, series: pd.Series) -> bool:
+        return pdt.is_numeric_dtype(series) and not pdt.is_bool_dtype(series)
+
+
+def is_date(series):
+    try:
+        _ = pd.to_datetime(series)
+        return True
+    except:
+        return False
+
+
+def to_date(series):
+    return pd.to_datetime(series)
+
+
+class DateTime(PandasProfilingBaseType):
+    @classmethod
+    def get_relations(cls):
+        return [
+            IdentityRelation(cls, Unsupported),
+            # TODO: optional
+            # TODO: coerce

```

Comment:
You can build a second typeset with the more expensive coercions for people to use or even expose it as a parameter passed to ProfileReport.